nixos/modules/services/databases/redis.nix

```diff
@@ -8,7 +8,6 @@ let
   condOption = name: value: if value != null then "${name} ${toString value}" else "";
 
   redisConfig = pkgs.writeText "redis.conf" ''
-    pidfile ${cfg.pidFile}
```

Comment:
Where will Redis write the PID file once this statement is removed?

Reply:
So far, we did ran redis in non-daemonized mode, meaning it didn't create a pidfile if not explicitly told so (and I removed the line in the version you specified).
In daemonized mode, it'll default to put it in `/run/redis.pid`, but won't complain if it can't at all.

> # If a pid file is specified, Redis writes it where specified at startup
> # and removes it at exit.
> #
> # When the server runs non daemonized, no pid file is created if none is
> # specified in the configuration. When the server is daemonized, the pid file
> # is used even if not specified, defaulting to "/var/run/redis.pid".
> #
> # Creating a pid file is best effort: if Redis is not able to create it
> # nothing bad happens, the server will start and run normally.

As we don't need the pidfile to manage the service, I removed the nixos module option.

I now hardcoded it to /run/redis/redis.pid (which is inside the RuntimeDirectory), enabled daemonizitation, and also enabled `Type=notify` support to signal readyness.